### PR TITLE
chore: Swap lifeCycleStatus cleanup

### PR DESCRIPTION
**Pull request by Zizzamia**
**What changed? Why?**

**Notes to reviewers**

**How has it been tested?**

### CODE REVIEW

File: src/swap/utils/getErrorMessage.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import {
```
Comment (cpcramer) [inline src/swap/utils/getErrorMessage.ts:1]: Nice! I like these getErrorMessage files

File: src/swap/components/SwapProvider.tsx [hunk 2]
```diff
@@ -139,13 +133,20 @@ export function SwapProvider({
 
         destination.setAmount(formattedAmount);
       } catch (err) {
-        handleError({ quoteError: err as SwapError });
+        setLifeCycleStatus({
+          statusName: 'error',
+          statusData: {
+            code: 'TmSPc01', // Transaction module SwapProvider component 01 error
```
Comment (cpcramer) [inline src/swap/components/SwapProvider.tsx:139]: Hmm I don't love this error code

Comment (Zizzamia) [reply src/swap/components/SwapProvider.tsx:139]: This is going to be the new starnard for all error code, where the goal is to create location from where the error is happening.

See error code as longitude and latitude.

So in this case TmSPc01 means
Tm = Transaction module
SPc = Swap Provider component
01 = First error of the file

Comment (Zizzamia) [reply src/swap/components/SwapProvider.tsx:139]: I am going to rewrite all the error code to be more like this, and instead have the Message and Error value be more about description on what's up.

Comment (Zizzamia) [reply src/swap/components/SwapProvider.tsx:139]: Have an error code as location, help debugging, and support developers when something is broken, as the first thing we will check, it's where the Error Code is originated.

Comment (cpcramer) [reply src/swap/components/SwapProvider.tsx:139]: Interesting, ok got it! Once all error codes match this format i'm sure I'll get used to it quickly

Comment (Zizzamia) [reply src/swap/components/SwapProvider.tsx:139]: cc @abcrane123 @cpcramer @0xAlec

**Review by cpcramer — APPROVED** [review]

File: src/swap/components/SwapProvider.test.tsx [hunk 3]
```diff
@@ -144,41 +172,6 @@ describe('SwapProvider', () => {
...
   });
 
-  it('should handle submit correctly', async () => {
-    const mockOnError = vi.fn();
-    const mockOnSuccess = vi.fn();
-    let _submitFunction: (
-      onError?: (error: SwapError) => void,
-      onSuccess?: (txReceipt: TransactionReceipt) => void | Promise<void>,
-    ) => void;
-    const TestComponent = () => {
```
Comment (Zizzamia) [inline src/swap/components/SwapProvider.test.tsx:154]: Should be only one TestComponent we use for various tests.

File: src/swap/components/SwapProvider.test.tsx [hunk 4]
```diff
@@ -326,29 +310,87 @@ describe('SwapProvider', () => {
...
+          .textContent,
+      ).toBe('error');
+      expect(
+        screen.getByTestId('context-value-lifeCycleStatus-statusData-code')
+          .textContent,
+      ).toBe('TmSPc02');
+    });
+  });
+
+  it('should setLifeCycleStatus to error when buildSwapTransaction returns an error', async () => {
```
Comment (Zizzamia) [inline src/swap/components/SwapProvider.test.tsx:377]: Adding more tests and re-organizing the tests as the order on where the code is located.

File: src/swap/components/SwapProvider.tsx [hunk 5]
```diff
@@ -46,28 +46,15 @@ export function SwapProvider({
...
+  const config = useConfig();
   const [loading, setLoading] = useState(false);
   const [isTransactionPending, setPendingTransaction] = useState(false);
   const [lifeCycleStatus, setLifeCycleStatus] = useState<LifeCycleStatus>({
     statusName: 'init',
     statusData: null,
   }); // Component lifecycle
-  const [error, setError] = useState<SwapErrorState>();
-
-  const handleError = useCallback(
```
Comment (Zizzamia) [inline src/swap/components/SwapProvider.tsx:57]: handleError is replaced with the setLifeCycleStatus

File: src/swap/types.ts [hunk 6]
```diff
@@ -72,7 +72,7 @@ export type GetSwapQuoteParams = {
 export type GetSwapQuoteResponse = SwapQuote | SwapError;
 
 export type GetSwapMessageParams = {
-  error?: SwapErrorState;
```
Comment (Zizzamia) [inline src/swap/types.ts:75]: Goodbye many type of errors, and we should have only one type.

File: src/swap/utils/getSwapMessage.test.ts [hunk 7]
```diff
@@ -93,7 +93,7 @@ describe('getSwapMessage', () => {
     expect(getSwapMessage(params2)).toBe(SwapMessage.FETCHING_QUOTE);
   });
 
-  test('returns INCOMPLETE_FIELD when required fields are missing', () => {
```
Comment (Zizzamia) [inline src/swap/utils/getSwapMessage.test.ts:96]: it should always be `it`, those `test` is probably some bad CoPilot advice.

File: vitest.config.ts [hunk 8]
```diff
@@ -23,10 +23,10 @@ export default defineConfig({
       ],
       reportOnFailure: true,
       thresholds: {
-        statements: 99.58,
-        branches: 99.56,
+        statements: 99.65,
```
Comment (Zizzamia) [inline vitest.config.ts:26]: Getting closer to 100%

**Pull request merged**